src/icmp/Icmp.h

```diff
@@ -36,14 +38,15 @@ struct pingerReplyData {
     pingerReplyData() { memset(&payload, 0, sizeof(payload)); }
     Ip::Address from;
     unsigned char opcode = '\0';
-    int rtt = 0;
+    int rtt = 0; // in milliseconds
     int hops = 0;
     int psize = 0;
     char payload[PINGER_PAYLOAD_SZ];
 };
 
 struct icmpEchoData {
-    struct timeval tv;
+    std::chrono::time_point<std::chrono::system_clock> tv;
+    static_assert(std::is_trivially_copyable_v<std::chrono::time_point<std::chrono::system_clock>>);
```

Comment:
Please move the assert to where the tested invariant is used.

If possible, I recommend testing the whole icmpEchoData structure instead of its individual data member for this assert. Doing so may also help with avoiding type duplication. If that duplication is still there, please name `std::chrono::time_point<std::chrono::system_clock>` type and reuse that name instead of spelling it out twice in this context.